proxy/tests/telemetry.rs

```diff
@@ -740,3 +743,54 @@ fn metrics_have_no_double_commas() {
     assert!(!scrape.contains(",,"), "outbound metrics had double comma");
 
 }
+
+// https://github.com/runconduit/conduit/issues/613
+#[test]
+#[cfg_attr(not(feature = "flaky_tests"), ignore)]
+fn metrics_compression() {
+    let _ = env_logger::try_init();
+
+    info!("running test server");
+    let srv = server::new().route("/hey", "hello").run();
+
+    let ctrl = controller::new()
+        .destination("tele.test.svc.cluster.local", srv.addr)
+        .run();
+    let proxy = proxy::new()
+        .controller(ctrl)
+        .inbound(srv)
+        .metrics_flush_interval(Duration::from_millis(500))
+        .run();
+    let client = client::new(proxy.inbound, "tele.test.svc.cluster.local");
+    let metrics = client::http1(proxy.metrics, "localhost");
+
+    let do_scrape = || {
+        let resp = metrics.request(
+            metrics.request_builder("/metrics")
+                .method("GET")
+                .header("Accept-Encoding", "gzip")
```

Comment:
What Accept-Encoding does Prometheus actually send? It would be good to test exactly what it sends + "deflate, gzip" + "gzip,deflate" + "brotli, gzip, deflate" for future-proofing.

Reply:
Prometheus' [documentation](https://prometheus.io/docs/instrumenting/exposition_formats/#format-variants-comparison) lists only "gzip". It's possible that the actual header differs from what's listed in the documentation --- could be investigated further.

Reply:
Alternatively, we could just unconditionally serve it compressed as that would avoid the complexity of parsing the Accept-Encoding header field completely. It's unlikely that Prometheus is ever going to be configured to NOT support gzip any time soon.

Reply:
This is a fair point. I felt it was better to be able to serve the metrics uncompressed as well, as the metrics endpoint is also often useful for manual debugging as well, and I thought getting useful information out of it should be as low-effort as possible. It also would mean we'll have to handle compression in all of our tests. However, I'd be happy to change this behaviour of you prefer. 

What do you think?

Reply:
Either we should make sure the Accept-Encoding parsing is robust or we should skip parsing it. I don't have a strong preference. My weak preference is to just test the Accept-Encoding more thoroughly.

Reply:
I think there is value in not sending compressed data optionally, so that someone can poke the endpoint without needing a gzip lib.